### ISSUE TITLE
Quantizing to int4 not working

### ISSUE THREAD

**Issue by basioli-k** [openvinotoolkit/nncf#2200]
**NNCF version: 2.6.0**

I am trying to quantize some layers of my model to `int4` to gain performance.
Here's an example of the code I am using:

```
import nncf
import torch
import numpy as np
from nncf.quantization.advanced_parameters import AdvancedQuantizationParameters, QuantizationParameters

class DummyDataset(torch.utils.data.Dataset):
    """ Loads images from a folder """
    def __init__(self, input_shapes, input_names):
        self.input_shapes = input_shapes
        self.input_names = input_names

    def __len__(self):
        return 1

    def __getitem__(self, index):
        return { self.input_names[i]: np.random.rand(*input_shape) for i, input_shape in enumerate(self.input_shapes) }

class DummyModel(torch.nn.Module):

    def __init__(self):
        super().__init__()
        self.conv = torch.nn.Conv2d(3, 6, 3, 1, 1)
        
    def forward(self, x):
        return self.conv(x)
    
def quantize_model(model, input_shapes, input_names):
    def transform_fn(data):
        data_dict = data
        return tuple(data_dict[key][0].to(torch.float32) for key in data_dict)
    
    dummy_dl = torch.utils.data.DataLoader(DummyDataset(input_shapes, input_names))
    calibration_dataset = nncf.Dataset(dummy_dl, transform_fn)
    
    advanced_parameters = AdvancedQuantizationParameters(activations_quantization_params=QuantizationParameters(8, "asymmetric", per_channel=False),
                                                         weights_quantization_params=QuantizationParameters(4, "symmetric"))
    
    return nncf.quantize(model, calibration_dataset, subset_size=1, preset=nncf.QuantizationPreset.MIXED, advanced_parameters=advanced_parameters)

def main():
    quantized_model = quantize_model(DummyModel(), [(1, 3, 256, 256)], ["x"])
    
    print(quantized_model._nncf.external_quantizers)
    
if __name__ == "__main__":
    main()
```

I would expect this to work based on the [docs](https://github.com/openvinotoolkit/nncf/blob/develop/docs/compression_algorithms/Quantization.md).

The result of this script is:
```
ValueError: Quantization parameter constraints specified in NNCF config are incompatible with HW capabilities as specified in HW config type 'CPU'.             First conflicting quantizer location: DummyModel/NNCFConv2d[conv]/conv2d_0
```

From looking into the problem I found the error occurs when comparing the quantization parameters to the values in [this directory](https://github.com/openvinotoolkit/nncf/tree/master/nncf/common/hardware/configs), where only the VPU seems to support `4` and `2` bit quantization.

Note that I tried this with `target_device=GPU` and `target_device=CPU`.

Is this expected behavior? If so, will `int4` support be available on the GPU anytime soon?

**Comment by ljaljushkin** [comment]
There's no support of int4 quantization on CPU and GPU. Currently, int4 is up-casted to int8 during the execution, and it shouldn't be faster than int8. 
However, there are some plans to support int4 weight compression for LLM.

**Comment by ljaljushkin** [comment]
> @ljaljushkin so the only hw supporting int4 is VPU so far?

@chamecall, yes and even VPU doesn't support execution of weight and activations in int4.

**Comment by ljaljushkin** [comment]
Just FYI, int4 weight compression has been merged - #2218. At least `INT4_SYM` mode should give some performance boost with OpenVINO from current master.

> Greeting, @basioli-k! There's no support of int4 quantization on CPU and GPU. Currently, int4 is up-casted to int8 during the execution, and it shouldn't be faster than int8. However, there are some plans to support int4 weight compression for LLM. Please refer to the API description: https://github.com/openvinotoolkit/nncf/blob/develop/docs/compression_algorithms/CompressWeights.md

@basioli-k, The original question was about 4-bit quantization of weight and activations. That is not going to be supported in the short term – therefore closing the issue.
